Sources/Classes/Search/BaseSearchViewController.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  Copyright (c) 2019 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import TableKit
+import RxSwift
+
+open class BaseSearchViewController<Item,
+    ItemViewModel,
+    ViewModel,
+    CustomView: UIView & TableViewHolder>: BaseCustomViewController<ViewModel, CustomView>
+where ViewModel: BaseSearchViewModel<Item, ItemViewModel> {
+
+    // MARK: - Properties
+
+    private let disposeBag = DisposeBag()
+    private let searchResultsViewController: UIViewController & SearchResultsViewController
+    private lazy var searchController = UISearchController(searchResultsController: searchResultsViewController)
+    private var didEnterText = false
+
+    // MARK: - Initialization
+
+    init(viewModel: ViewModel, searchResultsController: UIViewController & SearchResultsViewController) {
+        self.searchResultsViewController = searchResultsController
+        super.init(viewModel: viewModel)
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Life Cycle
+
+    open override func loadView() {
+        view = customView
```

Comment:
BaseCustomViewController should do it for you

Reply:
fixed